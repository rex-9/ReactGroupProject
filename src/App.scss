body {
  font-family: 'Quicksand', serif;
}

.menu {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 80px 20px 80px;
  margin-bottom: 20px;
  border-bottom: 1px solid gray;
}

.leftNav {
  display: flex;
  align-items: center;
}

.leftNav header {
  margin-left: 40px;
  font-size: 30px;
  font-weight: bold;
}

.stick {
  border-right: 1px solid black;
}

table {
  border: 2px solid black;
}

th {
  font-size: 20px;
  font-weight: bold;
  padding: 5px 0;
}

td {
  padding: 10px;
}

.profile {
  display: flex;
  justify-content: space-between;
  padding: 0 80px 0 80px;
}

.missionProfiles {
  width: 40%;
}

.missionProfiles h1 {
  font-size: 25px;
  font-weight: bold;
}

.missionProfiles div {
  border: 1px solid black;
  padding: 8px;
}

.rocketProfiles {
  width: 40%;
}

.rocketProfiles h1 {
  font-size: 25px;
  font-weight: bold;
}

.rocketProfiles div {
  border: 1px solid black;
  padding: 8px;
}
