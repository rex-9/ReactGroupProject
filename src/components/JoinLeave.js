import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { joinMission, leaveMission } from '../redux/mission';

const leaveStyle = {
  border: '1px solid #E87F88',
  borderRadius: '5px',
  padding: '3px 5px',
  textAlign: 'center',
  cursor: 'pointer',
  color: '#E87F88',
};

const joinStyle = {
  border: '1px solid #63686C',
  borderRadius: '5px',
  padding: '3px 5px',
  textAlign: 'center',
  cursor: 'pointer',
  color: '#63686C',
};

const Leave = (props) => {
  const { id } = props;
  const dispatch = useDispatch();
  const leave = (e) => {
    e.preventDefault();
    dispatch(leaveMission(id));
  };
  return (
    <button type="button" style={leaveStyle} onClick={leave} onKeyDown={leave}>Leave Mission</button>
  );
};

const Join = (props) => {
  const { id } = props;
  const dispatch = useDispatch();
  const join = (e) => {
    e.preventDefault();
    dispatch(joinMission(id));
  };
  return (
    <button type="button" style={joinStyle} onClick={join} onKeyDown={join}>Join Mission</button>
  );
};

Leave.propTypes = {
  id: PropTypes.string.isRequired,
};

Join.propTypes = {
  id: PropTypes.string.isRequired,
};

export { Join, Leave };
