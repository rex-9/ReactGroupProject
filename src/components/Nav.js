import { Link } from 'react-router-dom';
import { useState } from 'react';
import planet from '../assets/planet.png';

const style = {
  textDecoration: 'none', fontFamily: 'Montserrat', color: '#A7A7A7', letterSpacing: '1.9px',
};

const hoverStyle = {
  textDecoration: 'none', fontFamily: 'Montserrat', color: '#121212', letterSpacing: '1.9px',
};

const planetLogo = {
  backgroundColor: 'white', borderRadius: '50%', padding: '5px',
};

const Nav = () => {
  const [rocketsHover, setRocketsHover] = useState(false);
  const [missionsHover, setMissionsHover] = useState(false);
  const [myProfileHover, setMyProfileHover] = useState(false);

  const rocketsEnter = () => {
    setRocketsHover(true);
  };

  const rocketsLeave = () => {
    setRocketsHover(false);
  };

  const missionsEnter = () => {
    setMissionsHover(true);
  };

  const missionsLeave = () => {
    setMissionsHover(false);
  };

  const myProfileEnter = () => {
    setMyProfileHover(true);
  };

  const myProfileLeave = () => {
    setMyProfileHover(false);
  };

  return (
    <nav>
      <div id="menu">
        <header>Tesla</header>
        <ul>
          <li onMouseEnter={rocketsEnter} onMouseLeave={rocketsLeave}>
            <Link
              style={rocketsHover ? hoverStyle : style}
              to="/"
            >
              Rockets
            </Link>
          </li>
          <li onMouseEnter={missionsEnter} onMouseLeave={missionsLeave}>
            <Link
              style={missionsHover ? hoverStyle : style}
              to="/missions"
            >
              Missions
            </Link>
          </li>
          <li onMouseEnter={myProfileEnter} onMouseLeave={myProfileLeave}>
            <Link
              style={myProfileHover ? hoverStyle : style}
              to="/profile"
            >
              My Profile
            </Link>
          </li>
        </ul>
      </div>
      <img
        width="30px;"
        height="30px;"
        style={planetLogo}
        src={planet}
        alt=""
      />
    </nav>
  );
};
export default Nav;
