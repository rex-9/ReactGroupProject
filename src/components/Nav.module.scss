nav a.active {
  color: #fff;
  background: #7600dc;
  text-decoration: underline;
}

nav a.inactive {
  font-style: italic;
}

a {
  color: #545e6f;
  text-decoration: none;
}
