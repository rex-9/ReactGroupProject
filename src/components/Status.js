const activeStyle = {
  borderRadius: '5px',
  color: 'white',
  padding: '3px 10px',
  textAlign: 'center',
  backgroundColor: '#1BA3B8',
};

const inactiveStyle = {
  borderRadius: '5px',
  color: 'white',
  padding: '3px 10px',
  textTransform: 'uppercase',
  textAlign: 'center',
  backgroundColor: '#6D757D',
};

const ActiveStatus = () => (
  <div style={activeStyle}>
    Active Member
  </div>
);

const InactiveStatus = () => (
  <div style={inactiveStyle}>
    Not a member
  </div>
);

export { ActiveStatus, InactiveStatus };
