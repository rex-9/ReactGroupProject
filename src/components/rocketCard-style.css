.rocket {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  gap: 1rem;
}

.rocket .rocket__img {
  width: 300px;
}

.rocket__img img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.reserved {
  display: inline-block;
  background-color: #1ba3b8;
  padding: 2px 4px;
  margin-right: 10px;
  font-size: 14px;
  border-radius: 3px;
  color: #fff;
}
