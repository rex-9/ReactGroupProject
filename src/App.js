import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';

import Nav from './components/Nav';
import Rockets from './pages/Rockets';
import Missions from './pages/Missions';
import MyProfile from './pages/MyProfile';
import './App.scss';

const App = () => (
  <Router>
    <Nav />
    <Routes>
      <Route path="/" element={<Rockets />} />
      <Route path="/missions" element={<Missions />} />
      <Route path="/profile" element={<MyProfile />} />
    </Routes>
  </Router>
);

export default App;
