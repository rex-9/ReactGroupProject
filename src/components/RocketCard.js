import React from 'react';
import { useDispatch } from 'react-redux';
import './rocketCard-style.css';
import PropTypes from 'prop-types';
import { cancelReservation, reserveRocket } from '../redux/rocket';

const RocketCard = ({ rocketData }) => {
  const dispatch = useDispatch();
  const {
    id, description, name, reserved, image,
  } = rocketData;

  const handleReserve = (id) => {
    dispatch(reserveRocket(id));
  };

  const handleCancelReserve = (id) => {
    dispatch(cancelReservation(id));
  };

  return (
    <section className="rocket-card-component">
      <div className="rocket">
        <div className="left">
          <div className="rocket__img">
            <img src={image} alt="" />
          </div>
        </div>

        <div className="right">
          <h2>{name}</h2>
          <p>
            {reserved && <span className="reserved">Reserved</span>}
            {description}
          </p>

          {!reserved && (
            <button
              type="button"
              className="btn btn--primary"
              onClick={() => handleReserve(id)}
            >
              Reserve Rocket
            </button>
          )}

          {reserved && (
            <button
              type="button"
              className="btn btn--secondary"
              onClick={() => handleCancelReserve(id)}
            >
              Cancel Reservation
            </button>
          )}
        </div>
      </div>
    </section>
  );
};

RocketCard.protoTypes = {
  rocketData: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    reserved: PropTypes.bool,
    image: PropTypes.string,
    description: PropTypes.string,
  }),
};

export default RocketCard;
