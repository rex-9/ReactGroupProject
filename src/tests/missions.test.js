import React from 'react';
import { fireEvent, screen } from '@testing-library/react';
import renderWithProviders from './utils';
import App from '../App';
import { Join, Leave } from '../components/JoinLeave';
import { ActiveStatus, InactiveStatus } from '../components/Status';

describe('Render All of the components', () => {
  test('Render Pages', () => {
    renderWithProviders(<App />);

    expect(screen.getByText(/Space Traveler's Hub/i)).toMatchSnapshot(); // Nav component

    fireEvent.click(screen.getByText(/Missions/i));
    expect(screen.getByText(/Description/i)).toMatchSnapshot(); // Mission component

    expect(screen.getByText(/Description/i)).toMatchSnapshot(); // Mission component

    fireEvent.click(screen.getByText(/My Profile/i));
    expect(screen.getByText(/My Missions/i)).toMatchSnapshot(); // MyProfile component
  });

  test('Render Components', () => {
    renderWithProviders(<Join />);
    expect(screen.getByText(/Join Mission/i)).toMatchSnapshot(); // Join component
    renderWithProviders(<InactiveStatus />);
    expect(screen.getByText(/NOT A MEMBER/i)).toMatchSnapshot(); // Inactive component

    renderWithProviders(<ActiveStatus />);
    expect(screen.getByText('Active Member')).toMatchSnapshot(); // Active component
    renderWithProviders(<Leave />);
    expect(screen.getByText('Leave Mission')).toMatchSnapshot(); // Leave component
  });
});
