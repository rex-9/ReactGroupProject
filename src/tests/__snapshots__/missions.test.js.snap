// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render All of the components Render Components 1`] = `
<button
  style="border: 1px solid #63686c; border-radius: 5px; padding: 3px 5px; text-align: center; cursor: pointer; color: rgb(99, 104, 108);"
  type="button"
>
  Join Mission
</button>
`;

exports[`Render All of the components Render Components 2`] = `
<div
  style="border-radius: 5px; color: white; padding: 3px 10px; text-transform: uppercase; text-align: center; background-color: rgb(109, 117, 125);"
>
  Not a Member
</div>
`;

exports[`Render All of the components Render Components 3`] = `
<div
  style="border-radius: 5px; color: white; padding: 3px 10px; text-align: center; background-color: rgb(27, 163, 184);"
>
  Active Member
</div>
`;

exports[`Render All of the components Render Components 4`] = `
<button
  style="border: 1px solid #e87f88; border-radius: 5px; padding: 3px 5px; text-align: center; cursor: pointer; color: rgb(232, 127, 136);"
  type="button"
>
  Leave Mission
</button>
`;

exports[`Render All of the components Render Pages 1`] = `
<header>
  Space Traveler's Hub
</header>
`;

exports[`Render All of the components Render Pages 2`] = `
<th
  style="width: 50%;"
>
  Description
</th>
`;

exports[`Render All of the components Render Pages 3`] = `
<div>
  MyProfile
</div>
`;
